src/js/services/conversation-service.js

```diff
@@ -235,3 +246,28 @@ export function postPostback(actionId) {
         store.dispatch(showErrorNotification(store.getState().ui.text.actionPostbackError));
     });
 }
+
+
+export function fetchMoreMessages() {
+    const {conversation: {hasMoreMessages, messages}} = store.getState();
+
+    // TODO : add independent guard against parallel fetching
```

Comment:
Can't do that since you would effectively  be able to call it multiple time
in a row and it would be true... Just need a flag independent of the UI one

On Aug 18, 2016 15:44, "chloepouprom" <notifications@github.com> wrote:

> In src/js/services/conversation-service.js
> <https://github.com/smooch/smooch-js/pull/374#discussion_r75375663>:
>
> > @@ -235,3 +246,28 @@ export function postPostback(actionId) {
> >          store.dispatch(showErrorNotification(store.getState().ui.text.actionPostbackError));
> >      });
> >  }
> > +
> > +
> > +export function fetchMoreMessages() {
> > +    const {conversation: {hasMoreMessages, messages}} = store.getState();
> > +
> > +    // TODO : add independent guard against parallel fetching
>
> Hmmm I guess I can just check the appState to see if
> isFetchingMoreMessages is true. Since we set it right before calling it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/smooch/smooch-js/pull/374/files/81170bcbd64e9da1408e20b8b55912cd25985932#r75375663>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAvuFGOFLmVZKju1AXqNwYUT2rwExjkPks5qhLYBgaJpZM4Jm5kw>
> .
>